src/dolphin/gx/GXInit.c

```diff
@@ -572,17 +615,26 @@
 /* 8033AF34 00337B14  38 21 00 48 */	addi r1, r1, 0x48
 /* 8033AF38 00337B18  7C 08 03 A6 */	mtlr r0
 /* 8033AF3C 00337B1C  4E 80 00 20 */	blr 
+} // clang-format on
+#pragma pop
 
-# https://decomp.me/scratch/r2ALq // 7842 (85.20%)
-__GXInitGX:
+// https://decomp.me/scratch/r2ALq // 7842 (85.20%)
+#pragma push
+asm unk_t __GXInitGX()
+{ // clang-format off
+    nofralloc
 /* 8033AF40 00337B20  7C 08 02 A6 */	mflr r0
 /* 8033AF44 00337B24  90 01 00 04 */	stw r0, 4(r1)
 /* 8033AF48 00337B28  94 21 FF 88 */	stwu r1, -0x78(r1)
 /* 8033AF4C 00337B2C  93 E1 00 74 */	stw r31, 0x74(r1)
 /* 8033AF50 00337B30  93 C1 00 70 */	stw r30, 0x70(r1)
-/* 8033AF54 00337B34  80 82 E8 20 */	lwz r4, lbl_804DE200@sda21(r2)
-/* 8033AF58 00337B38  80 62 E8 24 */	lwz r3, lbl_804DE204@sda21(r2)
-/* 8033AF5C 00337B3C  80 02 E8 28 */	lwz r0, lbl_804DE208@sda21(r2)
+
+#pragma region wtf
+/* 8033AF54 00337B34  80 82 E8 20 */	lwz r4, -0x17E0(r2)
+/* 8033AF58 00337B38  80 62 E8 24 */	lwz r3, -0x17DC(r2)
+/* 8033AF5C 00337B3C  80 02 E8 28 */	lwz r0, -0x17D8(r2)
+#pragma endregion
```

Comment:
Does anyone know how to make inline asm consider an s32 as a variable instead of a literal, or whatever it's doing here?

Reply:
you should be able to use lwz r4, lbl_804DE200(r2) here if you declare lbl_804DE200 properly

Reply:
It's declared as `s32 const` because it needs to be in `sdata2`. But it seems the `const`-ness also inlines itself into the asm as an offset.

Reply:
![image](https://user-images.githubusercontent.com/5190964/188513635-87c5dfce-7ba7-4860-aaa9-dece198d298b.png)

Reply:
(On the left, dadosod dump of the original asm. On the right, dadosod dump of doing what you suggest.)

Reply:
```diff
@@ -22,7 +22,9 @@ volatile u32* __piReg;  // todo: OSPhysicalToUncached (0x0C003000)
 
 #pragma push
 #pragma force_active on
-s32 const lbl_804DE200 = 0x404040FF;
+s8 const lbl_804DE200[4] = {
+    64, 64, 64, -1
+};
 s32 const lbl_804DE204 = 0;
 s32 const lbl_804DE208 = -1;
 #pragma pop
@@ -630,7 +632,7 @@ asm unk_t __GXInitGX()
 /* 8033AF50 00337B30  93 C1 00 70 */   stw r30, 0x70(r1)
 
 #pragma region wtf
-/* 8033AF54 00337B34  80 82 E8 20 */   lwz r4, -0x17E0(r2)
+/* 8033AF54 00337B34  80 82 E8 20 */   lwz r4, lbl_804DE200(r2)
 /* 8033AF58 00337B38  80 62 E8 24 */   lwz r3, -0x17DC(r2)
 /* 8033AF5C 00337B3C  80 02 E8 28 */   lwz r0, -0x17D8(r2)
 #pragma endregion
```

Reply:
Huh, interesting. Nice work.